eventstore 3.5.0 -> 3.9.4

paullucas opened the pull request:
###### Motivation for this change
Need a more current version of EventStore (3.5.0 is outdated).

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

paullucas commented:
It seems to be working, but I did notice errors from the build.sh script. This is why I added git as a dependency (which may not have been necessary). Are the following errors expected?
```
Build succeeded.
	 0 Warning(s)
	 0 Error(s)

Time Elapsed 00:00:08.9089820
./build.sh: line 95: git: command not found
Reverted ./etc/EventStore.Rags.Tests/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./etc/EventStore.Documentation/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./etc/EventStore.UpgradeProjections/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.Padmin/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.ClientAPI/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.Projections.Core.Tests/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.Transport.Http/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.TestClient/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.ClusterNode.Web/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.Common/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.Core/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.BufferManagement.Tests/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/esquery/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.ClusterNode/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.Transport.Tcp/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.Core.Tests/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.Projections.Core/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.ClientAPI.Embedded/Properties/AssemblyInfo.cs
./build.sh: line 95: git: command not found
Reverted ./src/EventStore.BufferManagement/Properties/AssemblyInfo.cs
./build.sh: line 105: git: command not found
Reverted ./src/EventStore.Common/Utils/VersionInfo.cs
```

FRidh commented on pkgs/servers/nosql/eventstore/default.nix:
```diff
@@ -1,16 +1,16 @@
-{ stdenv, fetchFromGitHub, fetchpatch, mono, v8 }:
+{ stdenv, fetchFromGitHub, fetchpatch, mono, v8, git }:
```
why is git needed?

FRidh commented:
just adding git as function parameter won't add it to the buildInputs and thus won't solve the issue you report.

paullucas commented:
Thanks! I've now added Git to buildInputs. New error message: 
```
Build succeeded.
	 0 Warning(s)
	 0 Error(s)

Time Elapsed 00:00:10.2000840
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./etc/EventStore.Rags.Tests/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./etc/EventStore.Documentation/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./etc/EventStore.UpgradeProjections/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.Padmin/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.ClientAPI/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.Projections.Core.Tests/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.Transport.Http/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.TestClient/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.ClusterNode.Web/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.Common/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.Core/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.BufferManagement.Tests/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/esquery/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.ClusterNode/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.Transport.Tcp/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.Core.Tests/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.Projections.Core/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.ClientAPI.Embedded/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.BufferManagement/Properties/AssemblyInfo.cs
fatal: Not a git repository (or any of the parent directories): .git
Reverted ./src/EventStore.Common/Utils/VersionInfo.cs
```

paullucas commented:
@puffnfresh: Any ideas as to what's happening with the build.sh script?

a4d442663580 commented:
(@paullucas, the err is because `fetchFromGithub` afterward strips the repo (.git, ...) to make it deterministic (see https://github.com/NixOS/nixpkgs/blob/afec912d81e08339868e0e1e84234e6ed7f6b3fd/pkgs/build-support/fetchgit/nix-prefetch-git#L237) yet on the other hand build.sh depends on the git-related files)

paullucas commented:
I've been using this locally for months now and I haven't had any problems. If anyone is able to help with testing on MacOS / Linux, it should be good to go.

FRidh commented:
@paullucas judging from the `build.sh` and [this commit](https://github.com/EventStore/EventStore/commit/6d8c4fba29b99726154dff0db2c6402c51346f9b) the script tries to extract some info with git and update the `VersionInfo.cs` file. I have no idea whether it is relevant.

Because this patching is not working since we remove `.git`, and because you say the package works fine, I suggest you remove `git` again from the parameters and `buildInputs` and  just leave a note referring to this comment.

paullucas committed:
eventstore 3.5.0 -> 3.9.4
